File: zig/raindrops/Iteration1/raindrops.zig
const std = @import("std");

const Raindrop = struct {
    divisor: u3,
    sound: []const u8
};

const Raindrops = [3]Raindrop {
    .{ .divisor = 3, .sound = "Pling" },
    .{ .divisor = 5, .sound = "Plang" },
    .{ .divisor = 7, .sound = "Plong" },
};

pub fn convert(buffer: []u8, n: u32) []const u8 {
    var stream = std.io.fixedBufferStream(buffer);

    for (Raindrops) |raindrop| {
        if (n % raindrop.divisor == 0) {
            _ = stream.writer().write(raindrop.sound) catch unreachable;
        }
    }

    if (stream.pos == 0) {
        _ = stream.writer().print("{}", .{n}) catch unreachable;
    }

    return stream.getWritten();
}

File: nim/raindrops/Iteration1/raindrops.nim
proc convert*(n: int): string =
  var sound = ""

  if n mod 3 == 0:
    sound &= "Pling"
  if n mod 5 == 0:
    sound &= "Plang"
  if n mod 7 == 0:
    sound &= "Plong"

  if sound == "":
    return $n
  else:
    return sound
